lib/pbio/drv/ioport/ioport_lpf2.c

```diff
@@ -481,7 +481,20 @@ PROCESS_THREAD(pbdrv_ioport_lpf2_process, ev, data) {
     }
 
     while (true) {
-        PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_TIMER && etimer_expired(&timer));
+        PROCESS_WAIT_EVENT_UNTIL(ev == PROCESS_EVENT_POLL || (ev == PROCESS_EVENT_TIMER && etimer_expired(&timer)));
+
+        // If pbio_uartdev_process process polls us, we need to run dcm again,
+        // so set the iport state to disconnected.
+        if (ev == PROCESS_EVENT_POLL) {
```

Comment:
`PROCESS_EVENT_POLL` has special meaning, so this isn't the right event for an event that gets queued and takes a data argument. How about `PROCESS_EVENT_SERVICE_REMOVED`?

Reply:
Thanks, will try that.